apkid/rules/dex/obfuscators.yara

```diff
@@ -188,22 +188,73 @@ rule allatori_demo : obfuscator
     $s and is_dex
 }
 
-rule aamo : obfuscator
+rule aamo_str_enc_nop : obfuscator
 {
   meta:
-    description = "AAMO"
+    description = "AAMO (String decryption function + interleaved NOPs)"
     author = "P0r0"
     url = "https://github.com/necst/aamo"
-    backup_url = "https://github.com/P0r0/aamo"
     example1 = "c1ef860af0e168f924663630ed3b61920b474d0c8b10e2bde6bfd3769dbd31a8"
     example2 = "eb0d4e1ba2e880749594eb8739e65aa21b6f7b43798f04b6681065b396c15a78"
-    example3 = "b1e20bf3bdc53972424560e20c6d9ad12e5e47b8ed429a77f4ba5ff6cb92cb27"
-    example4 = "82a570c272579aacdc22410e152f4519738f4e0ececa84e016201c33ad871fa6"
 
   strings:
+    $opcodes = {
+        22 ?? ?? ?? 
+        ( 00 00 | 00 00 00 00 | 00 00 00 00 00 00 )
```

Comment:
Might there sometimes be 1 or 3 or 5 nops? Is it always in pairs?

Reply:
it is always in pairs

Reply:
sorry i misunderstood, i was talking about pairs of "00"s and not pairs of "nop" mnemonics.